### PR TITLE
[mle-router] fix `HasOneNeighborWithComparableConnectivity()`

### LINKED ISSUE #7612: Is the routerCount definition out of loop body?
https://github.com/openthread/openthread/blob/5c8a9cb5bc28ce6f794c0d8db7189b1f2dd586ba/src/core/thread/mle_router.cpp#L4307
The routerCount is always set to 0. The peerLinkQuality always get from the first one of Route64 TLV.

### PR DESCRIPTION
**Pull request by jwhui**
The `routerCount` value was being reset to zero with each iteration
over the router table. As a result, the comparisons were only
happening on the first entry in the received Route64 TLV.

Fixes #7612

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/core/thread/mle_router.cpp [hunk 1]
```diff
@@ -4298,66 +4298,56 @@ bool MleRouter::HasMinDowngradeNeighborRouters(void)
...
     for (Router &router : Get<RouterTable>().Iterate())
     {
-        uint8_t localLinkQuality = 0;
-        uint8_t peerLinkQuality  = 0;
-        uint8_t routerCount      = 0;
+        uint8_t localLinkQuality;
+        uint8_t peerLinkQuality;
 
-        if (router.GetRouterId() == mRouterId)
+        if (!router.IsStateValid() || (router.GetRouterId() == mRouterId) || (router.GetRouterId() == aRouterId))
```
Comment (simonlingoogle) [inline src/core/thread/mle_router.cpp:4310]:
```suggestion
        if (!router.IsStateValid() || router.GetRouterId() == mRouterId) || (router.GetRouterId() == aRouterId)
```
seems redundant parentheses?

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

I guess this issue could cause a Router to incorrectly downgrade if the Peer has a good connectivity to Router 0. 
Thoughts? @jwhui

**Comment by jwhui** [comment]
> I guess this issue could cause a Router to incorrectly downgrade if the Peer has a good connectivity to Router 0. Thoughts? @jwhui

Yes, the Thread Specification only allows a router to downgrade if it sees another neighbor with similar connectivity. Without this fix, the router may downgrade even if there is no neighbor with similar connectivity.

**Pull request merged**